interfaces/builtin/avahi_control.go

```diff
@@ -135,33 +138,27 @@ func (iface *avahiControlInterface) AppArmorConnectedPlug(spec *apparmor.Specifi
 }
 
 func (iface *avahiControlInterface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
-	if !release.OnClassic {
-		// NOTE: this is using avahi-observe permanent slot as it contains
-		// base declarations for running as the avahi service.
-		spec.AddSnippet(avahiObservePermanentSlotAppArmor)
-	}
+	// NOTE: this is using avahi-observe permanent slot as it contains
+	// base declarations for running as the avahi service.
+	spec.AddSnippet(avahiObservePermanentSlotAppArmor)
 	return nil
 }
 
 func (iface *avahiControlInterface) AppArmorConnectedSlot(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	if !release.OnClassic {
-		old := "###PLUG_SECURITY_TAGS###"
-		new := plugAppLabelExpr(plug)
-		// avahi-control implies avahi-observe, so add snippets for both here
-		snippet := strings.Replace(avahiObserveConnectedSlotAppArmor, old, new, -1)
-		spec.AddSnippet(snippet)
-		snippet = strings.Replace(avahiControlConnectedSlotAppArmor, old, new, -1)
-		spec.AddSnippet(snippet)
-	}
+	old := "###PLUG_SECURITY_TAGS###"
+	new := plugAppLabelExpr(plug)
+	// avahi-control implies avahi-observe, so add snippets for both here
+	snippet := strings.Replace(avahiObserveConnectedSlotAppArmor, old, new, -1)
+	spec.AddSnippet(snippet)
+	snippet = strings.Replace(avahiControlConnectedSlotAppArmor, old, new, -1)
+	spec.AddSnippet(snippet)
 	return nil
 }
 
 func (iface *avahiControlInterface) DBusPermanentSlot(spec *dbus.Specification, slot *snap.SlotInfo) error {
-	if !release.OnClassic {
-		// NOTE: this is using avahi-observe permanent slot as it contains
-		// base declarations for running as the avahi service.
-		spec.AddSnippet(avahiObservePermanentSlotDBus)
-	}
+	// NOTE: this is using avahi-observe permanent slot as it contains
+	// base declarations for running as the avahi service.
+	spec.AddSnippet(avahiObservePermanentSlotDBus)
```

Comment:
Likewise, the PermanentSlot policy for dbus should not be applied when the slotting snap is core, core18, etc. The deb/rpm will already have dbus policy on the system and we shouldn't try to override that.